src/docs/common_tasks/multiple_jvm_versions.md

```diff
@@ -0,0 +1,53 @@
+
+## Problem
+
+You want to compile and/or run different test or binary targets using different JVMs.
+
+Or, you need to specify command-line options globally for the JVM on a per-JVM version basis. For
+example, using `--add-opens` flags, but only when running on JVMs that support it.
+
+## Solution
+
+Declare JVM platforms using the `jvm-platform` configuration section to define the set of available
+platforms.
+
+On targets, add the kwarg `runtime_platform` set to the name of one of the platforms defined in
+`jvm-platform`. That will tell pants to use that platform when running that target in a JVM.
+
+To compile with a specific platform, set the `platform` on the target.
+
+
+## Discussion
+
+For compile time configuration, there are three main attributes, `source`, `target` and `args`.
+`source` and `target` map directly to the javac arguments of the same name, though they accept
+ more aliases than javac. `args`, allows the platform to specify additional, global to that
+ platform, compile arguments.
+
+For runtime configuration, `target` and `jvm_options` are used. `target` is used to determine the
+version of the JVM to use. `jvm_options` is used by some tasks to allow platforms to specify
+platform specific JVM options. These can make transitions between JVM versions smoother by
+allowing compatibility options to be provided globally.
+
+An example pants.toml config might look like this:
+
+    [jvm-platform]
+    default_platform: java8
+    default_runtime_platform: java10
+    platforms =
+    """
+     {
+        'java8': { 'source': '8', 'target': '8', 'args': [] },
+        'java10': {'source': '10', 'target': '10', 'args': [],
+                   'jvm_options': [
+                       # --add-opens is a 9+ flag
+                       '--add-opens=java.base/java.lang.reflect=ALL-UNNAMED'
+                   ] },
```

Comment:
How does it interact with the existing `test.junit.jvm_options`?

For example,
`./pants test --test-junit-jvm-options["-Dabc=123"] <target>`. Is the final jvm_option merged with `--add-opens` assuming the target is 10?

Reply:
The options are folded into the runtime ones, before the target options. https://github.com/pantsbuild/pants/pull/9150/files#diff-8603ca893539bade8d782040a56d85b8L411-R415

Reply:
So in that case, you'd end up with `--add-opens ... -Dabc=123`

Reply:
I see. Thanks for clarifying!

Reply:
Mentioning that explicitly in the docs would be good :)